tests/asynch/channel.rs

```diff
@@ -12,19 +12,41 @@ use std::collections::HashSet;
 use test_env_log::test;
 
 #[test]
-fn oneshot_once() {
+fn oneshot_once_blocking() {
     check_dfs(
         || {
             let (tx, rx) = oneshot::channel();
 
-            let _ = asynch::spawn(async move {
+            let send_task = asynch::spawn(async move {
                 tx.send(42u32).unwrap();
             });
 
             asynch::block_on(async {
                 let x = rx.await.unwrap();
                 assert_eq!(x, 42u32);
             });
+
+            // Don't drop send_task (causing it to finish)
```

Comment:
Not sure I follow why we have to do this. In the original version:
```
asynch::spawn(async move {
    tx.send(42u32).unwrap();
});

asynch::block_on(async {
    let x = rx.await.unwrap();
    assert_eq!(x, 42u32);
});
```
the sender handle gets dropped before `block_on`, but once we reach the `block_on`, we should be able to run the (now detached) sender task. So that test should work fine -- there's no deadlock because the sender task is still runnable.

Reply:
Yep totally right.